apps/betterangels/src/app/(private-screens)/add-note/Purpose/index.tsx

```diff
@@ -107,12 +105,9 @@ export default function Purpose(props: IPurposeProps) {
         <BasicInput
           placeholder="Enter purpose"
           maxLength={100}
-          onDelete={() => {
-            setValue('');
-            setErrors({ ...errors, purpose: true });
-          }}
+          onDelete={() => onChange('')}
```

Comment:
we should update the input type so that empty values are saves as `None` rather than `''`. 

in `apps/betterangels-backend/notes/types.py`, update `UpdateNoteInput`'s `purpose` field
```
...
class UpdateNoteInput:
    id: ID
    purpose: Optional[NonBlankString]
...
```